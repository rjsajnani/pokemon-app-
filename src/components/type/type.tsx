import { View, StyleSheet } from 'react-native';
import React from 'react';
import { Text } from '../common';
import { TypeStyle as styles } from './type.style';
import { TypeProps as Props } from './type.props';
import useCapitalizedString from '../../hooks/useCapitalizedString';
import { getPokemonTypeColor } from '../../constants';

export const Type: React.FC<Props> = (props) => {
  const { types } = props;
  const capitalizedString = useCapitalizedString();

  return (
    <View style={styles.content}>
      {types.map(({ slot, type: { name } }) => {
        return (
          <View
            key={slot}
            style={{
              ...styles.pill,
              backgroundColor: getPokemonTypeColor(name),
            }}
          >
            <Text style={styles.text} text={capitalizedString(name)} />
          </View>
        );
      })}
    </View>
  );
};
