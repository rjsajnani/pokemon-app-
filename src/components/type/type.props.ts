export interface PokemonType {
  slot: number;
  type: {
    name: string;
    url: string;
  };
}

export type TypeProps = {
  types: PokemonType[];
};
