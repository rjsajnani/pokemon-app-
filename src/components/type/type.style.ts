import { StyleSheet } from 'react-native';

export const TypeStyle = StyleSheet.create({
  content: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  pill: {
    paddingHorizontal: 30,
    backgroundColor: 'black',
    paddingVertical: 4,
    borderRadius: 20,
    marginHorizontal: 10,
  },
  text: {
    color: 'white',
  },
});
