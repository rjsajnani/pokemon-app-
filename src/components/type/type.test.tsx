import React from 'react';
import { render, RenderAPI } from '@testing-library/react-native';

import { Type } from './type';
import { PokemonType } from './type.props';

const dummyType: PokemonType[] = [
  { slot: 1, type: { name: 'electric', url: 'pokemon1-url' } },
  { slot: 2, type: { name: 'ground', url: 'pokemon2-url' } },
];

describe('<Type />', () => {
  let instance: RenderAPI;
  beforeEach(function () {
    instance = render(<Type types={dummyType} />);
  });

  it('renders correctly', () => {
    const tree = instance.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
