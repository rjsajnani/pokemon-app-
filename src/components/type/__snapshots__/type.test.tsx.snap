// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Type /> renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#F7D02C",
        "borderRadius": 20,
        "marginHorizontal": 10,
        "paddingHorizontal": 30,
        "paddingVertical": 4,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "black",
            "fontSize": 16,
            "letterSpacing": 0.2,
            "lineHeight": 24,
          },
          {
            "color": "white",
          },
        ]
      }
    >
      Electric
    </Text>
  </View>
  <View
    style={
      {
        "backgroundColor": "#E2BF65",
        "borderRadius": 20,
        "marginHorizontal": 10,
        "paddingHorizontal": 30,
        "paddingVertical": 4,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "black",
            "fontSize": 16,
            "letterSpacing": 0.2,
            "lineHeight": 24,
          },
          {
            "color": "white",
          },
        ]
      }
    >
      Ground
    </Text>
  </View>
</View>
`;
